### PR TITLE
Add replicate take 2

### PR DESCRIPTION
**Pull request by cbh123**
This PR adds a replicate integration to langchain. 

It's an updated version of https://github.com/hwchase17/langchain/pull/1993, but with updates to match latest replicate-python code. https://github.com/replicate/replicate-python.

### CODE REVIEW

File: docs/ecosystem/replicate.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Replicate
+This page covers how to use the Replicate ecosystem within LangChain.
+
+## Installation and Setup
+- Create a [Replicate](https://replicate.com) account. Get your api key and set it as an environment variable (`REPLICATE_API_TOKEN`)
+- Install the [Replicate python client](https://github.com/replicate/replicate-python) with `pip install replicate`
+
+## Calling a model
+
+Find a model on the [replicate explore page](https://replicate.com/explore), and then paste in the model name and version in this format: model_name/version
```
Comment (zeke) [inline docs/ecosystem/replicate.md:10]:
```suggestion
Find a model on the [Replicate explore page](https://replicate.com/explore), and then paste in the model name and version in this format: `owner-name/model-name:version`
```

File: docs/ecosystem/replicate.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+For example, for this [flan-t5 model]( https://replicate.com/daanelson/flan-t5), click on the API tab. The model name/version would be: `daanelson/flan-t5:04e422a9b85baed86a4f24981d7f9953e20c5fd82f6103b74ebc431588e1cec8`
```
Comment (zeke) [inline docs/ecosystem/replicate.md:12]:
```suggestion
For example, for this [flan-t5 model](https://replicate.com/daanelson/flan-t5), click on the API tab. The model name/version would be: `daanelson/flan-t5:04e422a9b85baed86a4f24981d7f9953e20c5fd82f6103b74ebc431588e1cec8`
```

File: docs/ecosystem/replicate.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Replicate
+This page covers how to use the Replicate ecosystem within LangChain.
```
Comment (zeke) [inline docs/ecosystem/replicate.md:2]:
```suggestion
This page covers how to run models on Replicate within LangChain.
```

File: docs/ecosystem/replicate.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+## Installation and Setup
+- Create a [Replicate](https://replicate.com) account. Get your api key and set it as an environment variable (`REPLICATE_API_TOKEN`)
```
Comment (zeke) [inline docs/ecosystem/replicate.md:5]:
```suggestion
- Create a [Replicate](https://replicate.com) account. Get your API key and set it as an environment variable (`REPLICATE_API_TOKEN`)
```

File: docs/ecosystem/replicate.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+- Install the [Replicate python client](https://github.com/replicate/replicate-python) with `pip install replicate`
+
+## Calling a model
+
+Find a model on the [Replicate explore page](https://replicate.com/explore), and then paste in the model name and version in this format: `owner-name/model-name:version`
+
+For example, for this [flan-t5 model](https://replicate.com/daanelson/flan-t5), click on the API tab. The model name/version would be: `daanelson/flan-t5:04e422a9b85baed86a4f24981d7f9953e20c5fd82f6103b74ebc431588e1cec8`
+
+Only the `model` param is required, but any other model parameters can also be passed in with the format input={model_param: value, ...}
```
Comment (zeke) [inline docs/ecosystem/replicate.md:14]:
```suggestion
Only the `model` param is required, but any other model parameters can also be passed in with the format `input={model_param: value, ...}`
```

File: docs/ecosystem/replicate.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+```python
+prompt = """
+Answer the following yes/no question by reasoning step by step.
+Can a dog drive a car?
+"""
+llm(prompt)
+```
+
+We can call any replicate model (not just LLMs) using this syntax. For example, we can call stable diffusion:
```
Comment (zeke) [inline docs/ecosystem/replicate.md:40]:
```suggestion
We can call any Replicate model (not just LLMs) using this syntax. For example, we can call stable diffusion:
```

**Review by hwchase17 — APPROVED** [review]
looks awsome - thanks!

**Pull request merged**